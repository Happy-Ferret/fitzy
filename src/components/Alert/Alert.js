import PropTypes from 'prop-types';
import React, { Fragment } from 'react';

import CloseIcon from '../../styles/icons/CloseIcon';
import { SRClose } from '../../styles/CloseButton.styled';
import { FitzyAlert, FitzyAlertIcon, FitzyClose } from './Alert.styled';

function Alert({ onCancel, use, className, children, title }) {
  return (
    <FitzyAlert role="alert" use={use} className={className}>
      <FitzyAlertIcon />
      <Fragment>
        <h2>{title}</h2>
        <div>{children}</div>
      </Fragment>
      {onCancel ? (
        <FitzyClose onClick={onCancel}>
          <CloseIcon height="12" width="12" />
          <SRClose>Close Alert</SRClose>
        </FitzyClose>
      ) : null}
    </FitzyAlert>
  );
}

Alert.propTypes = {
  /**
   * Title for the error message
   */
  title: PropTypes.string.isRequired,
  /**
   * Handler to close the Alert Banner
   */
  onCancel: PropTypes.func,
  /**
   * Type of warning show a different background for each
   */
  use: PropTypes.oneOf(['Error', 'Warning', 'Success', 'Info']),
  /**
   * Optional extra user supplied class name
   */
  className: PropTypes.string
};

Alert.defaultProps = {
  use: 'Error',
  className: ''
};

export default Alert;
