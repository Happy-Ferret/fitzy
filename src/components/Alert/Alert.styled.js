import styled from 'styled-components';

import { CloseButton } from '../../styles/helpers.styled';

const sizes = {
  Error: 'red',
  Warning: 'yellow',
  Success: 'green',
  Info: 'blue'
};

const FitzyAlert = styled.div`
  position: relative;
  color: white;
  padding: 1rem;
  background-color: ${({ type }) => sizes[type]};

  display: flex;
  align-items: baseline;
  flex-wrap: wrap;
  width: 100%;
  padding-right: 25px;
  border: 1px solid #d05353;
  background-color: #d97575;

  h2 {
    line-height: normal;
    font-weight: bold;
    margin-right: 1.5rem;
    margin-bottom: 0.25rem;
  }

  div {
    margin-top: 0.25rem;
  }
`;

const FitzyClose = CloseButton.extend`
  position: absolute;
  right: 1.5rem;
  top: 1.5rem;
  height: 0.75rem;
  width: 0.75rem;

  > svg {
    fill: currentColor;
  }

  > span {
  }
`;

/** @component */
export { FitzyAlert, FitzyClose };
