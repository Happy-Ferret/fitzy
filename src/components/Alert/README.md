Error Alert :

```jsx
initialState = { value: '' };

<Alert title="Authentication Error" onCancel={() => {}}>
  Oops, looks like an issue has occured.
</Alert>;
```

Warning Alert (without close button):

```jsx
initialState = { value: '' };

<Alert title="Trouble Ahead" level="Warning">
  Iceberg dead ahead.
</Alert>;
```

Success Alert :

```jsx
initialState = { value: '' };

<Alert title="Success !" level="Success" onCancel={() => {}}>
  This operation has succeeded.
</Alert>;
```
