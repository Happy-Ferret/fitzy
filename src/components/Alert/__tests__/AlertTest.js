import React from 'react';
import {
  // mountWithTheme,
  renderWithTheme
  // shallowWithTheme
} from '@sfitzpatrick/enzyme-context-helpers';

import Alert from '../Alert';

describe('Alert', () => {
  const defaultProps = {
    title: 'Login Error'
  };

  // let wrapper, mountedWrapper;

  beforeEach(() => {
    // wrapper = shallowWithTheme(<Alert {...defaultProps} />);
    // mountedWrapper = mountWithTheme(<Alert {...defaultProps} />);
  });

  describe('Alert', () => {
    test('should render correctly', () => {
      // console.log(wrapper.debug());
      // console.log(mountedWrapper.debug());
    });
  });

  describe('snapshots', () => {
    test('should match for basic', () => {
      const snap = renderWithTheme(
        <Alert {...defaultProps}>This is an error</Alert>
      ).toJSON();
      expect(snap).toMatchSnapshot();
    });

    test('should match with button', () => {
      const snap = renderWithTheme(
        <Alert {...defaultProps} onCancel={() => {}}>
          This is an error
        </Alert>
      ).toJSON();
      expect(snap).toMatchSnapshot();
    });
  });
});
