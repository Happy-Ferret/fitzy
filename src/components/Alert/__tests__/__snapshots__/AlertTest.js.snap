// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert snapshots should match for basic 1`] = `
.c2 {
  display: block;
  fill: currentColor;
  height: 24px;
  width: 24px;
}

.c2:hover {
  fill: #eee;
}

.c0 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  border-left: 5px solid hsl(0,57%,48%);
  box-shadow: 0 4px 6px hsla(0,0%,0%,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: hsl(0,57%,63%);
  color: #ffffff;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 1rem 2.5rem 1rem 3rem;
  position: relative;
  width: 100%;
}

.c0 h2 {
  font-weight: bold;
  margin-bottom: 0.25rem;
  margin-right: 1.5rem;
}

.c0 > div {
  margin-top: 0.25rem;
}

.c1 {
  left: 1rem;
  position: absolute;
  top: 1.125rem;
}

<div
  className="c0"
  role="alert"
>
  <svg
    className="c1 c2"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2a10 10 0 1 1 0 20 10 10 0 0 1 0-20zm0 2a8 8 0 1 0 0 16 8 8 0 0 0 0-16zm0 9a1 1 0 0 1-1-1V8a1 1 0 0 1 2 0v4a1 1 0 0 1-1 1zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
    />
  </svg>
  <h2>
    Login Error
  </h2>
  <div>
    This is an error
  </div>
</div>
`;

exports[`Alert snapshots should match with button 1`] = `
.c2 {
  display: block;
  fill: currentColor;
  height: 24px;
  width: 24px;
}

.c2:hover {
  fill: #eee;
}

.c4 {
  display: block;
  fill: currentColor;
  height: 12px;
  width: 12px;
}

.c4:hover {
  fill: #eee;
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: transparent;
  border: 0;
  color: inherit;
  cursor: pointer;
  line-height: 1;
  outline: none;
  padding: 0;
  height: 0.75rem;
  position: absolute;
  right: 1.5rem;
  top: 1.5rem;
  width: 0.75rem;
}

.c3::after {
  border-radius: 50%;
  content: '';
  left: 50%;
  padding: 1rem;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  -webkit-transition: background-color 0.25s ease-in;
  transition: background-color 0.25s ease-in;
}

.c3:hover::after {
  background-color: rgba(255,255,255,0.1);
}

.c5 {
  border: 0;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  border-left: 5px solid hsl(0,57%,48%);
  box-shadow: 0 4px 6px hsla(0,0%,0%,0.2);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: hsl(0,57%,63%);
  color: #ffffff;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 1rem 2.5rem 1rem 3rem;
  position: relative;
  width: 100%;
}

.c0 h2 {
  font-weight: bold;
  margin-bottom: 0.25rem;
  margin-right: 1.5rem;
}

.c0 > div {
  margin-top: 0.25rem;
}

.c1 {
  left: 1rem;
  position: absolute;
  top: 1.125rem;
}

<div
  className="c0"
  role="alert"
>
  <svg
    className="c1 c2"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 2a10 10 0 1 1 0 20 10 10 0 0 1 0-20zm0 2a8 8 0 1 0 0 16 8 8 0 0 0 0-16zm0 9a1 1 0 0 1-1-1V8a1 1 0 0 1 2 0v4a1 1 0 0 1-1 1zm0 4a1 1 0 1 1 0-2 1 1 0 0 1 0 2z"
    />
  </svg>
  <h2>
    Login Error
  </h2>
  <div>
    This is an error
  </div>
  <button
    className="c3"
    onClick={[Function]}
  >
    <svg
      className="c4"
      height="12"
      viewBox="0 0 212.982 212.982"
      width="12"
    >
      <path
        d="M131.804 106.491l75.936-75.936c6.99-6.99 6.99-18.323 0-25.312-6.99-6.99-18.322-6.99-25.312 0L106.491 81.18 30.554 5.242c-6.99-6.99-18.322-6.99-25.312 0-6.989 6.99-6.989 18.323 0 25.312l75.937 75.936-75.937 75.937c-6.989 6.99-6.989 18.323 0 25.312 6.99 6.99 18.322 6.99 25.312 0l75.937-75.937 75.937 75.937c6.989 6.99 18.322 6.99 25.312 0 6.99-6.99 6.99-18.322 0-25.312l-75.936-75.936z"
      />
    </svg>
    <span
      className="c5"
    >
      Close Alert
    </span>
  </button>
</div>
`;
